### DISCUSSION TITLE
adding-validation-to-strongly-typed-configuration-objects-in-asp-net-core/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#22] [Announcements]
# adding-validation-to-strongly-typed-configuration-objects-in-asp-net-core/

In this post I describe how to to verify your strongly typed configuration objects are correctly bound to your configuration at app startup

https://andrewlock.net/adding-validation-to-strongly-typed-configuration-objects-in-asp-net-core/

**Comment by RaushniGupta** [comment]
Nice work, I really appreciate for your sharing. Waiting for your next post. Well done! Visit Us: <a href="https://delhitourismgov.blogspot.com/">Delhi Tourism Blog</a>

**Comment by RaushniGupta** [comment]
It's actually a nice and helpful piece of info. I am happy that you simply shared this useful info with us. Please stay us up to date like this. Thanks for sharing. And you can get amazing article at. <a href="https://www.atlastravel.in/kashmir-tour-packages/">Kashmir Tour Packages from Mumbai</a>

**Comment by RaushniGupta** [comment]
Hii,
This is Great Post for me. now, i’m Learned from your article. thanks for sharing this kinda information’s for me. i glad with your Article!. <a href="https://indiatrekkiing.blogspot.com/">India Trekking Blog</a>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at July 11 2018, 10:11, in reply to [Muhammad Rehan Saeed](https://disqus.com/by/RehanSaeedUK/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/22#discussioncomment-6046047)</em>

---

<p>Done!</p>

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at July 11 2018, 10:45, in reply to [Muhammad Rehan Saeed](https://disqus.com/by/RehanSaeedUK/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/22#discussioncomment-6046047)</em>

---

<p>Though it's important to bear in mind the constraints of the approach I described  I suspect those constraints mean it won't find its way into the framework. The most important is that using a startup filter means validation occurs _once_ before the app starts. That isn't compatible with the configuration-reloading available with IOptionsMonitor for example. I also haven't looked into how it works with named options. I don't think either of those will be an issue for most people, but they're probably sufficiently inconsistent with existing features that it won't make it into the framework. We'll see!</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at September 29 2019, 01:37, in reply to [bit bonk](https://disqus.com/by/bitbonk/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/22#discussioncomment-6046059)</em>

---

<p>Approaches I've taken are to expose and endpoint that dumps the whole of configuration. But I wouldn't include that in production, as that could expose all sorts of secrets!</p>

[a comment by an automated account is not shown]

**Comment by rinomau** [comment]
Hi Andrew,
thanks for you excellent article.

Is it possible to validate the registered object before it is called with builder.build() so that it can be used as a dependency in a IServiceCollection extension method ?

I have
```
builder.Services.Configure<WebApiConfiguration>(builder.Configuration);
builder.Services.AddSingleton(resolver => resolver.GetRequiredService<Appsettings>().WebApiConfiguration);
builder.Services.AddSingleton<IValidatable>(resolver => resolver.GetRequiredService<Appsettings>().WebApiConfiguration);
```

and then I try to do 

```
var sc = builder.Configuration.GetSection(ServicesConfiguration.AppsettingsKey).Get<ServicesConfiguration>();
builder.Services.AddApplicationCore(sc);
```

But if the ServicesConfiguration is invalid the application crashes before passing to the validato

thanks !

**Reply by andrewlock** [reply]
Hi, it's not possible to do it _automatically_ I'm afraid, no. You would have to bind and validate it "manually" (by calling `Validate()` or similar yourself

[By the way, I have a newer version of this post now](https://andrewlock.net/adding-validation-to-strongly-typed-configuration-objects-in-dotnet-6/) 🙂

**Comment by amiradeli** [comment]
Thanks for sharing Andrew. Just a question. I know it's a bit opinionated but just want to know which all these validations, do we again need to do the check on properties when they are actually used in the code? From Defensive Programming point-of-view it's good to check them. On the other hand, it's kind of doing the same thing twice.

**Reply by amiradeli** [reply]
which all these validations => with all these validations :)
